### DISCUSSION TITLE
Field group interitance/product type inheritance structure

### DISCUSSION THREAD

**Discussion by nitech** [craftcms/cms#8999] [Ideas]
I find myself creating a lot of Product Types that share a lot of the same fields. I typically structure the fields this way: 

![image](https://user-images.githubusercontent.com/1279436/107017405-5fd22280-679f-11eb-97ff-736dba0212ca.png)

All the fields in the leftmost "Generelt" group are equal accross all my Product Types. The fields in "Detaljer" are specific for this particular product type. 

If I had the ability to structure the product types, I would perhaps structure them like this: 

- Electric Appliance
  - Sewing Machines
    - Overlock
    - Safety
- Light
  - Table Lights
  - Ceiling Lights

Each level in the product type hierarchy would have its own fields that would be propagated town through the hierarchy. I believe this is how most larger webshops structure their fields. Here is an example from komplett.no (google translated)

As you see, they first output a "Generally" group which to me seems to be the fields belonging specifically to the product type "Coffee Brewers" (with the possible exception of Warranty and Custom Standards"

I skip the next groups until I get to "Power supply" - which is a typical group you need accross any electric appliance. When we get down to `Sizes and Weights`, that's a group all of my products would need. Same goes for `Sizes and Weights (shipping)`

![screencapture-translate-google-translate-2021-02-05-10_53_19](https://user-images.githubusercontent.com/1279436/107018499-cb68bf80-67a0-11eb-82de-bb416311cb27.png)

If you don't think this is the best way to solve the challenge, please explain the way you would solve it. Also - if I am simply "pushing the envelope" of Craft Commerce, please also tell me. I am not experienced with PIM-systems, but I believe this is a typical feature of a PIM-system.

**Comment by pdaleramirez** [comment]
@nitech Unfortunately there is no hierarchy between product types yet. I can see how this would be a great feature to have. We'll keep this open and see if we can include this feature on major releases.

For now to achieve it would require a custom plugin or module. It would need a relationship tables between the product types. Then query product types based on their relation or groupings.

**Comment by lukeholder** [comment]
Transferring this issue to https://github.com/craftcms/cms since field layouts are a core feature of Craft. 

Let's open the discussion up there on sharing field tabs across field layouts or some sort of field layout inheritance.

**Comment by nitech** [comment]
Related to https://github.com/craftcms/commerce/issues/2059

**Comment by nitech** [comment]
@brandonkelly What does it mean that you closed this issue without futher comment? Does it mean that you reject the idea?

**Reply by pixelmachine** [reply]
All feature requests have been moved to the discussions area, presumably to clear up the issues and allow for voting on ideas.
